tests/helm/common_test.go

```diff
@@ -475,6 +475,15 @@ func GetTolerations(object unstructured.Unstructured) ([]corev1.Toleration, erro
 	return nil, nil
 }
 
+func ContainsImagePullSecret(imagePullSecrets []corev1.LocalObjectReference, expectedSecret string) bool {
```

Comment:
Shouldn't a Pod Spec have one imagePullSecret, why is this a list? We might be able to avoid a loop here, if we only pass in the imagePullSecret Key.

Reply:
yeah i noticed there was one image pull secret being used in this whole repo, but i saw the [documentation](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#containers) which supports a list

```
ImagePullSecrets is an optional list of references to secrets in the 
same namespace to use for pulling any of the images used by this PodSpec.
```

i would say keeping a loop is just better in the long term if in the far future there will be more than one secret being used.

i wouldn't be opposed to avoiding a loop now too, since the code would be cleaner 😆

Reply:
Customers may use multiple different repositories for different pod and it is more usable to apply all secrets in one place

Reply:
Could we test for two values?

Reply:
yep